File: Binary Search/0274.H-Index/Solution.java
class Solution {
    public int hIndex(int[] citations) {
        int left = 0, right = citations.length;
        while (left < right) {
            int mid = (left + right + 1) >> 1;
            if (check(citations, mid)) {
                left = mid;
            } else {
                right = mid - 1;
            }
        }
        return left;
    }

    public boolean check(int[] nums, int x) {
        int cnt = 0;
        for (int v : nums) {
            if (v >= x) {
                cnt++;
            }
        }
        return cnt >= x;
    }
}

File: Binary Search/0274.H-Index/Solution.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        def check(x):
            cnt = 0
            for v in citations:
                if v >= x:
                    cnt += 1
            return cnt >= x

        left, right = 0, 1000
        while left < right:
            mid = (left + right + 1) >> 1
            if check(mid):
                left = mid
            else:
                right = mid - 1
        return left